### ISSUE TITLE
megacheck: panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by adamdecaf** [dominikh/go-tools#344]
Hi! 

Running the latest commit (as of writing) panics. We haven't really narrowed down what code it could be, but I've tried on Go 1.11 on and off the GOPATH. I've reproduced this on Windows 10 Pro and WSL and a co-worker originally had the problem. 

```
$ git log -n1
commit 88497007e8588ea5b6baee991f74a1607e809487
Author: Dominik Honnef <dominik@honnef.co>
Date:   Sat Jul 28 08:38:16 2018 +0200

    staticcheck: remove redundant case
```

```
$ megacheck
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x661586]

goroutine 544 [running]:
honnef.co/go/tools/lint.(*Job).Errorf(0xc00be96dc0, 0x9cbc80, 0xc008d06700, 0x9589e9, 0x4a, 0x0, 0x0, 0x0, 0x0)
        /home/adam/code/src/honnef.co/go/tools/lint/lint.go:525 +0xc6
honnef.co/go/tools/staticcheck.(*Checker).CheckDiffSizeComparison(0xc000108680, 0xc00be96dc0)
        /home/adam/code/src/honnef.co/go/tools/staticcheck/lint.go:1186 +0x536
honnef.co/go/tools/staticcheck.(*Checker).CheckDiffSizeComparison-fm(0xc00be96dc0)
        /home/adam/code/src/honnef.co/go/tools/staticcheck/lint.go:257 +0x34
honnef.co/go/tools/lint.(*Linter).Lint.func1(0xc0070cfa80, 0xc0084f5140, 0xc00be96dc0)
        /home/adam/code/src/honnef.co/go/tools/lint/lint.go:425 +0x93
created by honnef.co/go/tools/lint.(*Linter).Lint
        /home/adam/code/src/honnef.co/go/tools/lint/lint.go:419 +0x2437
```

**Comment by dominikh** [comment]
Is the code that this is happening on publicly available?

**Comment by adamdecaf** [comment]
Yea, sorry forgot to link: https://github.com/moov-io/x9

**Comment by dominikh** [comment]
Can reproduce, will fix.

**Comment by dominikh** [comment]
FYI, it's caused by a bug in `(*Reader).parseLine` in your code base. `r.line[:1]` is of length 1, but the cases check for string values of length 2. 

Staticcheck crashes trying to report the error, since it isn't handling switch statements correctly in that particular check.

**Comment by dominikh** [comment]
Closed by e3ad64cb4ed3e25a30bf42def711f9cb5b004f72

**Comment by adamdecaf** [comment]
Perfect, thanks!
